### PR TITLE
Add lesson learned from polyfill.io

### PR DESCRIPTION
**Pull request by david-a-wheeler**

### CODE REVIEW

File: docs/Concise-Guide-for-Developing-More-Secure-Software.md [hunk 1]
```diff
@@ -33,5 +33,6 @@ Here is a concise guide for all software developers for secure software developm
 24. **Continuously improve**. Improve scores, look for tips, & apply as appropriate.
 25. **Manage succession**. Have clear governance & work to add active, trustworthy maintainer(s).
 26. **Prefer memory-safe languages**. Many vulnerabilities involve memory safety. Where practical, use memory-safe programming languages (most are) and keep memory safety enabled. Otherwise, use mechanisms like extra tools and peer review to reduce risk.
+27. **Ensure production websites only load from your own domains**. _Linking_ to other domains is fine, but don't load information from other domains such as JavaScript, CSS, and images. If you do, that domain's subversion might also subvert your site. See the [subverted polyfill.io revealed in 2024](https://blog.qualys.com/vulnerabilities-threat-research/2024/06/28/polyfill-io-supply-chain-attack).
```
Comment (ctcpip) [inline docs/Concise-Guide-for-Developing-More-Secure-Software.md:36]:
```suggestion
27. **Ensure production websites only load assets from your own domains**. _Linking_ to other domains is fine, but don't load assets from other domains such as JavaScript, CSS, and media. If you do, that domain's subversion might also subvert your site. See the [subverted polyfill.io revealed in 2024](https://blog.qualys.com/vulnerabilities-threat-research/2024/06/28/polyfill-io-supply-chain-attack).
```

also, should the text reference the term supply chain attack or similar?

Comment (ctcpip) [inline docs/Concise-Guide-for-Developing-More-Secure-Software.md:36]: should the text reference the term supply chain attack or similar?

Comment (david-a-wheeler) [reply docs/Concise-Guide-for-Developing-More-Secure-Software.md:36]: I like the reword, thanks!

**Comment by ctcpip** [comment]
> I wouldn't say "supply chain" on just this item. There are many supply chain attacks, and this is only one kind. If we single this out with the term "supply chain", I fear people might think they're identical.

I actually backtracked on that and had removed my comment, but I guess there was a race condition :)

**Review by ctcpip — APPROVED** [review]

**Comment by david-a-wheeler** [comment]
@ctcpip - no problem! I merged in your improvement, & made a few more tweaks. I think it's better, thank you!

**Review by ljharb — APPROVED** [review]
i thought we all learned this 10-20 years ago, but apparently not :-p

**Comment by david-a-wheeler** [comment]
[ljharb](https://github.com/ljharb) said:
> i thought we all learned this 10-20 years ago, but apparently not :-p

Hehe. I think the problem is that new people keep showing up, and instead of learning from the past, they presume that the past has nothing to teach them. The past has very brutal ways to show otherwise :-(.

**Review by SecurityCRob — APPROVED** [review]
+1

**Comment by drusso-rh** [comment]
+1

**Pull request merged**